src/Processing.js

```diff
@@ -4066,10 +4066,36 @@
     p.noLoop = function() {
       doLoop = false;
       loopStarted = false;
-      clearInterval(looping);
+      
+      looping = false;
       curSketch.onPause();
     };
 
+requestAnimationFrame= (function(){
+      return window.requestAnimationFrame ||
+             window.webkitRequestAnimationFrame ||
+             window.mozRequestAnimationFrame    ||
+             function( callback ){
+                var _h= function(){callback(Date.now());};
+                window.setTimeout(_h, curMsPerFrame);
+             };
+     })();
+
+    var _now, _then=0, _doDraw = function(t) {
+      requestAnimationFrame(_doDraw);
```

Comment:
While this makes sense from a timing perspective, having this scheduling happen first means that if there are errors in the code that gets called during onframestart, redraw, and/or onframeend, we keep trying to run it over and over. If we want to keep the timing stable, we'll need to a try/catch around the onframe/redraw/onframe block so that if there's a throw anywhere, we catch it, and at the very least set `looping` to false before throwing the exception on.

However, this function *keeps running* even if looping is set to false, because the scheduling happens prior to the loop check. That's probably going to cause unnecessary work for the browser, so it's probably worth also switching L4085 and L4086 so that we first check whether to do an early return because we shouldn't be looping, and *then* request the operation for the new frame, then process the current one.